transport/src/main/java/io/netty/channel/EventLoopGroups.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.channel;
+
+import io.netty.util.internal.SystemPropertyUtil;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * {@link EventLoopGroup} utilities.
+ */
+public final class EventLoopGroups {
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(EventLoopGroups.class);
+
+    protected static final String CHANNEL_PACKAGE = SystemPropertyUtil.get(
+            "io.netty.packagePrefix", EventLoopGroups.class.getPackage().getName());
+
+    private static final List<String> STANDARD_IMPLEMENTATIONS = Collections.unmodifiableList(Arrays.asList(
+            "nio", "oio"));
+
+    private static final List<String> NATIVE_IMPLEMENTATIONS = Collections.unmodifiableList(Arrays.asList(
+            "epoll"
+    ));
+
+    private static final List<Constructor<? extends EventLoopGroup>> EVENT_LOOP_CONSTRUCTORS =
+            createEventLoopGroupConstructors();
+
+    private EventLoopGroups() {
+    }
+
+    /**
+     * Returns list of standard netty event loop group implementations.
+     *
+     * @return list of standard implementations.
+     */
+    protected static List<String> standardImplementations() {
+        return STANDARD_IMPLEMENTATIONS;
+    }
+
+    /**
+     * Returns list of native implementations.
+     *
+     * @return list of native implementations.
+     */
+    protected static List<String> nativeImplementations() {
+        return NATIVE_IMPLEMENTATIONS;
+    }
+
+    /**
+     * Returns fastest non-blocking {@link EventLoopGroup} instance for running platform. Tries to initialize native
+     * implementation for running platform (epoll on Linux for example) first and fallbacks to
+     * {@link io.netty.channel.nio.NioEventLoopGroup} if native implementation for running platform is not available.
+     *
+     * @return event loop group
+     */
+    public static EventLoopGroup fastestNonBlocking() {
```

Comment:
`fastest` can mean a lot of things, IMO we should avoid using this terminology. What about `newEventLoopGroup()`?